A possibly unreachable branch in royin.py

In royin.py:

I suppose the last else block is unreachable since word[i] == _RO_RUA should be always true in this whole block.

```Python
elif word[i] == _RO_RUA and i + 1 < len(word):
    if word[i + 1] == _RO_RUA:
        word = list(word)
        del word[i + 1]
        if i + 2 == len_cons:
            word[i] = "an"
        else:
            word[i] = "a"
        word = "".join(word)
        i += 1
    elif word[i] == _RO_RUA:
        word = word.replace(
            consonants[i], _CONSONANTS[consonants[i]][1]
        )
        i += 1
    else:
        word = word.replace(
            consonants[i],
            _CONSONANTS[consonants[i]][1]
        )
        i += 1
```

I ran python -m unittest discover -s tests -p 'test_transliterate.py' on commit 3ea5f5e43b0f354bc8e27a1cb5928e75489be6af. It didn't pass two tests. 

Did I do it properly, for example, correct commit/branch? Should I just ignore these two tests for now?

```
..FF...
======================================================================
FAIL: test_romanize_royin_basic (test_transliterate.TestTransliteratePackage)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vee/Develop/free/pythainlp/tests/test_transliterate.py", line 63, in test_romanize_royin_basic
    self.assertEqual(romanize(word, engine="royin"), expect)
AssertionError: 'nppn' != 'nopphon'
- nppn
+ nopphon
?  +  ++


======================================================================
FAIL: test_romanize_royin_consistency (test_transliterate.TestTransliteratePackage)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vee/Develop/free/pythainlp/tests/test_transliterate.py", line 67, in test_romanize_royin_consistency
    self.assertEqual(
AssertionError: 'takpai' != 'takbai'
- takpai
?    ^
+ takbai
?    ^


----------------------------------------------------------------------
Ran 7 tests in 0.350s

FAILED (failures=2)
```

> 
> 
> I ran python -m unittest discover -s tests -p 'test_transliterate.py' on commit [3ea5f5e](https://github.com/PyThaiNLP/pythainlp/commit/3ea5f5e43b0f354bc8e27a1cb5928e75489be6af). It didn't pass two tests.
> 
> Did I do it properly, for example, correct commit/branch? Should I just ignore these two tests for now?
> 
> ```
> ..FF...
> ======================================================================
> FAIL: test_romanize_royin_basic (test_transliterate.TestTransliteratePackage)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/vee/Develop/free/pythainlp/tests/test_transliterate.py", line 63, in test_romanize_royin_basic
>     self.assertEqual(romanize(word, engine="royin"), expect)
> AssertionError: 'nppn' != 'nopphon'
> - nppn
> + nopphon
> ?  +  ++
> 
> 
> ======================================================================
> FAIL: test_romanize_royin_consistency (test_transliterate.TestTransliteratePackage)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/vee/Develop/free/pythainlp/tests/test_transliterate.py", line 67, in test_romanize_royin_consistency
>     self.assertEqual(
> AssertionError: 'takpai' != 'takbai'
> - takpai
> ?    ^
> + takbai
> ?    ^
> 
> 
> ----------------------------------------------------------------------
> Ran 7 tests in 0.350s
> 
> FAILED (failures=2)
> ```

OK. I fixed.

--- a/pythainlp/transliterate/royin.py
+++ b/pythainlp/transliterate/royin.py
@@ -167,10 +167,12 @@ def _replace_consonants(word: str, consonants: str) -> str:
                     consonants[0], _CONSONANTS[consonants[0]][0]
                 )
                 i += 1
-        elif i == len_word and consonants[i] == _RO_RUA and word[i - 1] == _RO_RUA:
-            word = word.replace(
-                    consonants[i], _CONSONANTS[consonants[i]][1]
-            )
+        elif (
+            i == len_word
+            and consonants[i] == _RO_RUA
+            and word[i - 1] == _RO_RUA
+        ):
+            word = word.replace(consonants[i], _CONSONANTS[consonants[i]][1])
         elif i < len_word and consonants[i] == _RO_RUA:
             if i + 1 == len_word and word[i] == _RO_RUA:
                 word = word.replace(
@@ -193,14 +195,12 @@ def _replace_consonants(word: str, consonants: str) -> str:
                     i += 1
                 else:
                     word = word.replace(
-                        consonants[i],
-                        _CONSONANTS[consonants[i]][1]
+                        consonants[i], _CONSONANTS[consonants[i]][1]
                     )
                     i += 1
             else:
                 word = word.replace(
-                    consonants[i],
-                    _CONSONANTS[consonants[i]][1]
+                    consonants[i], _CONSONANTS[consonants[i]][1]
                 )
                 i += 1
         else:
@@ -212,10 +212,6 @@ def _replace_consonants(word: str, consonants: str) -> str:
 
 # support function for romanize()
 def _romanize(word: str) -> str:
-    """
-    :param str word: a Thai word, should have already been tokenized.
-    :return: Spells out how the Thai word should be pronounced.
-    """
     word = _replace_vowels(_normalize(word))
     consonants = _RE_CONSONANT.findall(word)
 
@@ -231,13 +227,14 @@ def _romanize(word: str) -> str:
 
 
 def romanize(text: str) -> str:
-    """
-    Rendering Thai words in the Latin alphabet or "romanization",
-    using the Royal Thai General System of Transcription (RTGS),
-    which is the official system published by the Royal Institute of Thailand.
+    """Render Thai words in Latin alphabet, using RTGS
 
-    :param str text: Thai text to be romanized
-    :return: A string of Thai words rendered in the Latin alphabet.
+    Royal Thai General System of Transcription (RTGS),
+    is the official system by the Royal Institute of Thailand.
+
+    :param text: Thai text to be romanized
+    :type text: str
+    :return: A string of Thai words rendered in the Latin alphabet
     :rtype: str
     """
     words = word_tokenize(text)

--- a/pythainlp/transliterate/royin.py
+++ b/pythainlp/transliterate/royin.py
@@ -196,11 +196,6 @@ def _replace_consonants(word: str, consonants: str) -> str:
                         _CONSONANTS[consonants[i]][1]
                     )
                     i += 1
-            elif word[i] == _RO_RUA:
-                word = word.replace(
-                    consonants[i], _CONSONANTS[consonants[i]][1]
-                )
-                i += 1
             else:
                 word = word.replace(
                     consonants[i],

--- a/pythainlp/transliterate/royin.py
+++ b/pythainlp/transliterate/royin.py
@@ -152,10 +152,11 @@ def _replace_consonants(word: str, consonants: str) -> str:
 
     if len(consonants) == 1:
         return word.replace(consonants[0], _CONSONANTS[consonants[0]][0])
+    len_cons = len(consonants)
 
     i = 0
-    len_cons = len(consonants)
     while i < len_cons:
+        len_word = len(word)
         if i == 0:
             if consonants[0] == _HO_HIP:
                 word = word.replace(consonants[0], "")
@@ -166,16 +167,16 @@ def _replace_consonants(word: str, consonants: str) -> str:
                     consonants[0], _CONSONANTS[consonants[0]][0]
                 )
                 i += 1
-        elif consonants[i] == _RO_RUA and i == len(word) and word[i - 1] == _RO_RUA:
+        elif i == len_word and consonants[i] == _RO_RUA and word[i - 1] == _RO_RUA:
             word = word.replace(
                     consonants[i], _CONSONANTS[consonants[i]][1]
             )
-        elif consonants[i] == _RO_RUA and i < len(word):
-            if i + 1 == len(word) and word[i] == _RO_RUA:
+        elif i < len_word and consonants[i] == _RO_RUA:
+            if i + 1 == len_word and word[i] == _RO_RUA:
                 word = word.replace(
                     consonants[i], _CONSONANTS[consonants[i]][1]
                 )
-            elif word[i] == _RO_RUA and i + 1 < len(word):
+            elif i + 1 < len_word and word[i] == _RO_RUA:
                 if word[i + 1] == _RO_RUA:
                     word = list(word)
                     del word[i + 1]

